src/core/MOM_barotropic.F90

```diff
@@ -1052,6 +1057,30 @@ subroutine btstep(U_in, V_in, eta_in, dt, bc_accel_u, bc_accel_v, forces, pbce,
     wt_v(i,J,k) = CS%frhatv(i,J,k) * visc_rem
   enddo ; enddo ; enddo
 
+  if (CS%wt_uv_fix) then
+    do j=js,je ; do I=is-1,ie ; Iwt_u_tot(I,j) = wt_u(I,j,1) ; enddo ; enddo
+    do k=2,nz ; do j=js,je ; do I=is-1,ie
+      Iwt_u_tot(I,j) = Iwt_u_tot(I,j) + wt_u(I,j,k)
+    enddo ; enddo ; enddo
+    do j=js,je ; do I=is-1,ie
+      Iwt_u_tot(I,j) = G%mask2dCu(I,j) * Adcroft_reciprocal(Iwt_u_tot(I,j))
```

Comment:
Doing a function call to another module at every point in the loop is probably going to be expensive.  Doing an Adcroft reciprocal here is so simple that we should just add the logic to do it here - namely replace line 1066 with `if (abs(Iwt_u_tot(I,j)) > 0.0 ) Iwt_u_tot(I,j)= G%mask2dCu(I,j) / Iwt_u_tot(I,j)`, and analogously on line 1077.

Reply:
Fixed now. I took the liberty of squashing the change with the previous commit to keep the history concise. Let me know if there is a problem.